src/main/kotlin/com/amazon/elasticsearch/replication/action/replay/TransportReplayChangesAction.kt

```diff
@@ -178,7 +179,13 @@ class TransportReplayChangesAction @Inject constructor(settings: Settings, trans
         val options = IndicesOptions.strictSingleIndexNoExpandForbidClosed()
         val getMappingsRequest = GetMappingsRequest().indices(leaderIndex).indicesOptions(options)
         val getMappingsResponse = remoteClient.suspending(remoteClient.admin().indices()::getMappings, injectSecurityContext = true)(getMappingsRequest)
-        val mappingSource = getMappingsResponse.mappings().get(leaderIndex).get(type).source().string()
+        val indexMapping = getMappingsResponse.mappings().get(leaderIndex)
+        if (indexMapping.get(type) == null) {
+            log.error("Mapping response: $getMappingsResponse")
+            throw MappingNotAvailableException("Mapping for the index $leaderIndex is not available")
+        }
+        val mappingSource = indexMapping.get(type).source().string()
```

Comment:
Let us use safe-call so we are covered if we run into null at any nesting level. (https://kotlinlang.org/docs/reference/null-safety.html)

```kotlin
val mappingSource = getMappingsResponse?.mappings()?.get(leaderIndex)?.get(type)?.source()?.string()
if null == mappingSource {
    log
    throw MappingNotA..
}
```

Reply:
Updated this.